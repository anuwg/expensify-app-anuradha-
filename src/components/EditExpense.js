import React from 'react'
import {BrowserRouter, Route, Switch, Link, NavLink} from 'react-router-dom'
import {connect} from 'react-redux'
import EditForm from '../components/addAndEditForm'
import {editExpense} from '../actions/expense'
import {removeExpense} from '../actions/expense'

export class EditExpense extends React.Component{
    onSubmit=()=>{
        (expense)=>{
            props.editExpense(expense.id, expense)
            this.props.history.push('/')
           }
    }
    onClick=()=>{
        ()=>{
            props.removeExpense({id:this.expense.id})
           
           this.props.history.push('/')
       }
    }
    render(){
        return (
        <div>
            <p>this is the {this.props.match.params.id} page</p>
               <EditForm onSubmit={this.onSubmit} expense={this.props.expense}/>
        
               <button onClick={this.onClick}>Remove</button>
        </div>)
        
            
    }
}


const mapDispatchToProps=(dispatch)=>{
    return {
        editExpense:dispatch(editExpense(props.match.params.id, expense)),
        removeExpense:dispatch(removeExpense({id:props.match.params.id}))
    }

}
const mapStateToProps = (state, props)=>{
      return{
          expense:state.expenses.find((expense)=>expense.id===props.match.params.id)
      }
     }
    export default connect(mapStateToProps, mapDispatchToProps)(EditExpense)