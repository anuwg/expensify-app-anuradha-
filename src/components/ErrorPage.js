import React from 'react'
import {BrowserRouter, Route, Switch, Link, NavLink} from 'react-router-dom'
const ErrorPage=()=>(
    <div>
    <p>404 error</p>
    <Link to="/"> Go To Main Page</Link>
    </div>
    ) 

export default ErrorPage