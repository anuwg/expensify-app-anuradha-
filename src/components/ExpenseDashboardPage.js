import React from 'react'
import {BrowserRouter, Route, Switch, Link, NavLink} from 'react-router-dom'
import ExpenseList from './expenseList'
import InputForm from '../components/ExpenseListForm'
const ExpenseDashboardPage=()=>(
    <div>
    <p>this is the dashboard page</p>
    <InputForm/>
    <ExpenseList/>
    </div>
    )
    
    

export default ExpenseDashboardPage