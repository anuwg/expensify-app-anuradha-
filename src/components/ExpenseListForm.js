import React from 'react'
import {connect} from 'react-redux'
import {sortByDate, sortByAmount, setStartDate, setEndDate} from '../actions/filters'
import editFilter, { editFiler } from '../actions/filters'
import {DateRangePicker} from 'react-dates'
class InputForm extends React.Component{
    state={
        calenderFocused:null
    }

    onDatesChange=({ startDate, endDate })=>{
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate))
    }
    onFocusChange=(focusedInput)=>{
    this.setState(()=>({  calenderFocused:focusedInput}))
    }
    render(){
        return (
            <div>
            <input type="text" value={this.props.filters.text} onChange={(e)=>{
             this.props.dispatch(editFiler({text:e.target.value}))
            }}></input>
            <select value ={this.props.filters.sortBy} onChange={(e)=>{
                //e.target.value
                
                e.target.value==='date'?this.props.dispatch(sortByDate()):this.props.dispatch(sortByAmount())
            }}>
            <option value ="date">Date</option>
            <option value = "amount">Amount</option>
            </select>

            <DateRangePicker
            startDate={this.props.filters.startDate} 
            //startDateId="your_unique_start_date_id" 
            endDate={this.props.filters.endDate} 
           // endDateId="your_unique_end_date_id" 
            onDatesChange={this.onDatesChange}
            focusedInput={this.state.calenderFocused}
            onFocusChange={this.onFocusChange}
            showClearDates={true}
            numberOfMonths={1}
            isOutsideRange={()=>
                false
            }
            
            />
            </div>
        )
    }

}


const mapStateProps= (state)=>{
return {
    filters:state.filters
}
}
export default connect(mapStateProps)(InputForm)