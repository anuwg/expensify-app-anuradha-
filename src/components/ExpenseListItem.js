import React from 'react'
import {connect} from 'react-redux'
import {removeExpense} from '../actions/expense'
import {BrowserRouter, Route, Switch, Link, NavLink} from 'react-router-dom'
const Item = ({dispatch, expenseType, amount, createdAt,id})=>(

     <div>
     <Link to={`/edit/${id}`}><h3>{expenseType}</h3></Link>
     
     <p>{createdAt} - {amount}</p>
     
     </div>
 
)
// const mapStateProps= (state)=>{
//      return {
//          expenses:state.expenses
//      }
//      }
export default Item