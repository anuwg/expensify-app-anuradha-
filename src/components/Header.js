import React from 'react'
import {BrowserRouter, Route, Switch, Link, NavLink} from 'react-router-dom'
const  Header =()=>(
    <header>
    <h1>Expensify</h1>
    <NavLink activeClassName="is-active" to="/" exact={true}> Home Pageu </NavLink>
    
    <NavLink activeClassName="is-active" to="/create"> Create</NavLink>
    <NavLink activeClassName="is-active" to="/help">  Help </NavLink>
    </header>
)

export default Header