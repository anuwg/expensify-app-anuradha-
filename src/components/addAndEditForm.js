import React from 'react'
import moment from 'moment'
import {SingleDatePicker} from 'react-dates'

const date = moment()
console.log(date.format('MMM Do, YYYY'))
export default class FormChange extends React.Component{
    constructor(props){
        super(props)
        this.state={
        expenseType: props.expense?props.expense.expenseType:'',
        description:props.expense?props.expense.description:'',
        amount: props.expense?(props.expense.amount/100).toString():'',
        createdAt: props.expense?moment(props.expense.createdAt):moment(),
        focused:false,
    error:''}
    }
    
     valueChange=(e)=>{
        const item =e.target.value
      this.setState(()=>{
         
         return {expenseType:item}
     })
     }
     textChange=(e)=>{
         const item = e.target.value
         this.setState(()=>{
             return {description:item}
         })
     }
     amountChange=(e)=>{
      const amount = e.target.value
      if(!amount ||amount.match(/^\d{1,}(\.\d{0,2})?$/)){
          this.setState(()=>({amount}))
      }
     }
     dateChange=(createdAt)=>{
         if(createdAt){
            this.setState(()=>({createdAt}))
         }
         
     }
     changeFocus=({focused})=>{
  this.setState(()=>{
      return {focused}
  })
     }

     onSubmit=(e)=>{
 e.preventDefault()
  if(!this.state.expenseType ||!this.state.amount){
    this.setState(()=>({error:'Provide amount and type'}))
  } else{
      this.setState(()=>({error:''}))
      this.props.onSubmit({
        expenseType:this.state.expenseType,  
          amount:parseFloat(this.state.amount,10)*100,
          
          description:this.state.description,
          createdAt:this.state.createdAt.valueOf()
      })
  }
     }
 render(){
     
     return (
         <div>
         <p>{this.state.error}</p>
         <form onSubmit={this.onSubmit}>
         <input type='text' placeholder="expenseType" autoFocus value={this.state.expenseType}
          onChange={this.valueChange}
         
         />
         <input type='text' placeholder="amount" value={this.state.amount} onChange={this.amountChange} />
         
         <SingleDatePicker
          date={this.state.createdAt}
          onDateChange={this.dateChange}
          focused={this.state.focused}
          onFocusChange={this.changeFocus}
          numberOfMonths={1}
          isOutsideRange={()=> false}
         />
         <textarea placeholder="optional description" value ={this.state.description} onChange={this.textChange}></textarea>
         <button>Add Expense</button>
         </form>
         </div>
     )
 }
}