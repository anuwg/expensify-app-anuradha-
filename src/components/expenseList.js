import React from 'react'
import {connect} from 'react-redux'
import Item from './ExpenseListItem'
import SelectItems from '../selectors/expense'
 export const ExpenseList= (props)=>(
    <div>
    <h1>Expense-List</h1>
    {props.expense.length>0? props.expense.map((expense)=>{
        return <Item key={expense.id} {...expense}/>
    }) :<p>No items</p>}
   
   
    </div>
)



const connectFunction= (state)=>{
 return {
    expense:SelectItems(state.expenses, state.filters)
    
    
 }
}
const StoreDetails = connect(connectFunction)(ExpenseList)
export default StoreDetails