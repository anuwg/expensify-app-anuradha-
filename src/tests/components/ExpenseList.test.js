import React from 'react'
import {shallow} from 'enzyme'
//import toJSON from 'enzyme-to-json'
import {ExpenseList} from '../../components/expenseList'
import moment from 'moment'
const expenses = [{
    id:'1',
    expenseType:'rent',
    amount:1890,
    description:'',
    createdAt:0
},
{
    id:'2',
    expenseType:'travel',
    amount:18,
    description:'',
    createdAt:moment(0).subtract(5,'days').valueOf()
},{
    id:'3',
    expenseType:'school-expense',
    amount:180988,
    description:'',
    createdAt:moment(0).add(5,'days').valueOf()
}]
test('items present',()=>{
    const wrapper = shallow(<ExpenseList expense={expenses}/>)
    expect(wrapper).toMatchSnapshot()
})
test('items present',()=>{
    const wrapper = shallow(<ExpenseList expense={[]}/>)
    expect(wrapper).toMatchSnapshot()
})
