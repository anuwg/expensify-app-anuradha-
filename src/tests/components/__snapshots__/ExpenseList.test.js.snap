// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`items present 1`] = `
<div>
  <h1>
    Expense-List
  </h1>
  <Item
    amount={1890}
    createdAt={0}
    description=""
    expenseType="rent"
    id="1"
    key="1"
  />
  <Item
    amount={18}
    createdAt={-432000000}
    description=""
    expenseType="travel"
    id="2"
    key="2"
  />
  <Item
    amount={180988}
    createdAt={432000000}
    description=""
    expenseType="school-expense"
    id="3"
    key="3"
  />
</div>
`;

exports[`items present 2`] = `
<div>
  <h1>
    Expense-List
  </h1>
  <p>
    No items
  </p>
</div>
`;
